src/API/Google/AdsCampaign.php

```diff
@@ -329,6 +339,65 @@ public function delete_campaign( int $campaign_id ): int {
 		}
 	}
 
+	/**
+	 * Retrieves the status of converting campaigns.
+	 * The status is cached for an hour during unconverted.
+	 *
+	 * - unconverted    - Still need to convert some older campaigns
+	 * - converted      - All campaigns are converted to PMax campaigns
+	 * - not-applicable - User never had any older campaign types
+	 *
+	 * @since x.x.x
+	 *
+	 * @return string
+	 */
+	public function get_campaign_convert_status(): string {
+		$convert_status = $this->options->get( OptionsInterface::CAMPAIGN_CONVERT_STATUS );
+
+		if ( ! is_array( $convert_status ) || empty( $convert_status['status'] ) ) {
+			$convert_status = [ 'status' => 'unconverted' ];
+		}
+
+		// Refetch if status is unconverted and older than an hour.
+		if (
+			'unconverted' === $convert_status['status'] &&
+			( empty( $convert_status['updated'] ) || time() - $convert_status['updated'] > HOUR_IN_SECONDS )
+		) {
+			$old_campaigns         = 0;
+			$old_removed_campaigns = 0;
+
+			try {
+				foreach ( $this->get_campaigns( false, false ) as $campaign ) {
+					if ( CampaignType::PERFORMANCE_MAX !== $campaign['type'] ) {
+						if ( CampaignStatus::REMOVED === $campaign['status'] ) {
+							$old_removed_campaigns++;
+						} else {
+							$old_campaigns++;
+						}
+					}
+				}
+			} catch ( Exception $e ) {
+				// Error when retrieving campaigns, do not handle conversion.
+				$convert_status['status'] = 'not-applicable';
```

Comment:
I was originally thinking the worst case scenario would be that those users don't see the update notifications, which isn't going to block anything for them. But I suppose we can introduce a 4th status `unknown`, which will also retry after the cache expires.

Reply:
>I was originally thinking the worst case scenario would be that those users don't see the update notifications

Yes, it will only affect the notifications and renaming of the old campaigns. 

> But I suppose we can introduce a 4th status unknown, which will also retry after the cache expires.

Yes, or we just leave the status as `unconverted`. But I am not sure if this could cause that we end in an infinite loop if the get_campaigns() fails constantly. What do you think?

Reply:
Already added the `unknown` status. An infinite loop would be pretty rare, and then it would still cache it for an hour.
If we want we can always show the notice if it's unknown or unconverted. But since it will retry after an hour and the notices run once a day, I think it will eventually show up with just unconverted. Unless they are really unlucky that they can't retrieve any campaigns, but then they have a greater problem to fix and we don't necessarily need to be prompting them that their campaigns are going to update.